src/core/Global.ttslua

```diff
@@ -838,15 +859,23 @@ function onClick_toggleOption(_, id)
   applyOptionPanelChange(id, state)
 end
 
+-- called by the language selection dropdown
+function languageSelected(_, selectedIndex, id)
+  optionPanel[id] = tonumber(selectedIndex)
+end
+
 -- sets the option panel to the correct state (corresponding to 'optionPanel')
 function updateOptionPanelState()
   for id, enabled in pairs(optionPanel) do
     if (type(enabled) == "boolean" and enabled)
         or (type(enabled) == "string" and enabled)
         or (type(enabled) == "table" and #enabled ~= 0) then
-      self.UI.setAttribute(id, "isOn", true)
+      UI.setAttribute(id, "isOn", true)
+    -- hard-coded updating of the language selection dropdown
+    elseif id == "cardLanguage" and type(enabled) == "number" then
```

Comment:
For clarity in a conditional block it's usually better to but more specific branches first, and more general ones later.